section_faiss_vector_index.go

```diff
@@ -63,10 +63,17 @@ func (v *faissVectorIndexSection) AddrForField(opaque map[int]resetable, fieldID
 
 // metadata corresponding to a serialized vector index
 type vecIndexMeta struct {
-	startOffset int
-	indexSize   uint64
-	vecIds      []int64
-	deleted     []int64
+	startOffset       int
+	indexSize         uint64
+	vecIds            []int64
+	deleted           []int64
+	indexOptimizedFor string
+}
+
+// maps the index optimization type int to the string
+var indexOptimizationMap = map[int]string{
```

Comment:
Not a good idea to add this reverse lookup here, let's keep it all in one place - within bleve_index_api.